direct/core/src/main/java/cz/o2/proxima/direct/commitlog/CommitLogReaders.java

```diff
@@ -129,14 +130,12 @@ public boolean hasExternalizableOffsets() {
 
     @Getter private final ThroughputLimiter limiter;
     private final Collection<Partition> partitions;
-    private final int numPartitions;
     private long watermark = Long.MIN_VALUE;
 
     public LimitedCommitLogReader(CommitLogReader delegate, ThroughputLimiter limiter) {
       super(delegate);
-      this.limiter = Objects.requireNonNull(limiter);
+      this.limiter = SerializableUtils.clone(Objects.requireNonNull(limiter));
```

Comment:
why is this necessary? either way, this should be caller's responsibility to provide healthy limiter instance

Reply:
Hmm... This looks strange. My intent definitely was to clone this into the `LogObserver`. This looks like a bug.

Reply:
Okay, definitely a bug. The clone should have been in different place. Fixed.